src/codeu/chat/client/simplegui/OAuth2.java

```diff
@@ -0,0 +1,149 @@
+import com.google.api.client.util.store.FileDataStoreFactory;
+import com.google.api.services.oauth2.Oauth2;
+import com.google.api.services.oauth2.model.Tokeninfo;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Command-line sample for the Google OAuth2 API described at <a
+ * href="http://code.google.com/apis/accounts/docs/OAuth2Login.html">Using OAuth 2.0 for Login
+ * (Experimental)</a>.
+ *
+ * @author Yaniv Inbar
+ */
+public class OAuth2 {
+
+  /**
+   * Be sure to specify the name of your application. If the application name is {@code null} or
+   * blank, the application will log a warning. Suggested format is "MyCompany-ProductName/1.0".
+   */
+  private static final String APPLICATION_NAME = "CodeU";
+
+  /**
+   * Directory to store user credentials.
+   */
+  private static final java.io.File DATA_STORE_DIR =
+      new java.io.File(System.getProperty("user.home"), ".store/oauth2");
+
+  /**
+   * Global instance of the {@link DataStoreFactory}. The best practice is to make it a single
+   * globally shared instance across your application.
+   */
+  private static FileDataStoreFactory dataStoreFactory;
+
+  /**
+   * Global instance of the HTTP transport.
+   */
+  private static HttpTransport httpTransport;
+
+  /**
+   * Global instance of the JSON factory.
+   */
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+  /**
+   * OAuth 2.0 scopes.
+   */
+  private static final List<String> SCOPES = Arrays.asList(
+      "https://www.googleapis.com/auth/userinfo.profile",
+      "https://www.googleapis.com/auth/userinfo.email");
+
+  private static Oauth2 oauth2;
+  private static GoogleClientSecrets clientSecrets;
+
+  /**
+   * Authorizes the installed application to access user's protected data.
+   */
+  private static Credential authorize() throws Exception {
+    // load client secrets
+    clientSecrets = GoogleClientSecrets.load(JSON_FACTORY, new FileReader("client_secrets.json"));
+    if (clientSecrets.getDetails().getClientId().startsWith("Enter")
+        || clientSecrets.getDetails().getClientSecret().startsWith("Enter ")) {
+      System.out.println("Enter Client ID and Secret from https://code.google.com/apis/console/ "
+          + "into oauth2-cmdline-sample/src/main/resources/client_secrets.json");
+      System.exit(1);
+    }
+    // set up authorization code flow
+    GoogleAuthorizationCodeFlow flow = new GoogleAuthorizationCodeFlow.Builder(
+        httpTransport, JSON_FACTORY, clientSecrets, SCOPES).setDataStoreFactory(
+        dataStoreFactory).build();
+    // authorize
+    return new AuthorizationCodeInstalledApp(flow, new LocalServerReceiver()).authorize("user");
+  }
+
+  /**
+   * Authorizes and returns email after logging in.
+   *
+   * @return email
+   */
+  public static String getEmail() {
+    try {
+      httpTransport = GoogleNetHttpTransport.newTrustedTransport();
```

Comment:
More advanced topic, since I don't know if you're familiar with concurrency.

getEmail stores the transport into httpTransport, which is a class field. Imagine two threads call getEmail() concurrently, the following sequence could happen:
Thread 1: create transport 1
Thread 1: store transport 1 into httpTransport
Thread 2: create transport 2
Thread 2: store transport 2 into httpTransport
Thread 1: pass httpTransport to Oauth2.Builder()
Thread 2: pass httpTransport to Oauth2.Builder()
Without knowing anything about GoogleNetHttpTransport, at this point the state of httpTransport is probably confused.

IT's possible that GoogleNetHttpTransport is designed to work in this case, but then it'd be clearer to simply declare httpTransport locally in this method and avoid the possible concurrency issues and address any doubts the reader may have.